redshells/model/graph_convolutional_matrix_completion.py

```diff
@@ -152,17 +99,32 @@ def _adjust_adjacency_matrix(self, normalization_type):
         else:
             raise ValueError(f'normalization_type must be "left", "right" or "symmetric", but {normalization_type} is passed.')
 
-    @staticmethod
-    def _feature_layer(hidden_size: int, input_data):
-        x = tf.keras.layers.Dense(hidden_size, use_bias=False, activation=None, kernel_initializer='glorot_normal')(input_data)
-        return x
+    def _add_user_feature(self, encoder_hidden_size, encoder_size, side_initial="glorot_normal"):
+        if len(self.input_user_features) != 0:
+            self.user_feature_layers = [self._feature_convert_layer(encoder_hidden_size, kernel_initializer=layer_init) for layer_init in self.gcmc_user_feature_layers_init]
+            x = tf.reduce_prod([layer(feature)+1.0 for layer, feature in zip(self.user_feature_layers, self.input_user_features)], axis=0)
+            self.user_side_info_layer = tf.keras.layers.Dense(encoder_size, use_bias=False, activation='relu', kernel_initializer=side_initial)
+            x = tf.gather(self.user_side_info_layer(x), self.input_user_feature_indices)
+            if self.user_encoder is None:
+                self.user_encoder = x
+            else:
+                self.user_encoder += x  # TODO: discussion for simple addition
```

Comment:
@mski-iksm 
Is it OK for 'TODO' to be still remaining?

Reply:
I've resolved the TODOs.